src/cmd/linuxkit/run_qemu.go

```diff
@@ -509,11 +525,15 @@ func buildQemuCmdline(config QemuConfig) (QemuConfig, []string) {
 		qemuInitrdPath := config.Path + "-initrd.img"
 		qemuArgs = append(qemuArgs, "-kernel", qemuKernelPath)
 		qemuArgs = append(qemuArgs, "-initrd", qemuInitrdPath)
-		cmdlineString, err := ioutil.ReadFile(config.Path + "-cmdline")
-		if err != nil {
+		if cmdline, err := ioutil.ReadFile(config.Path + "-cmdline"); err != nil {
 			log.Errorf("Cannot open cmdline file: %v", err)
 		} else {
-			qemuArgs = append(qemuArgs, "-append", string(cmdlineString))
+			cmdlineString := string(cmdline)
+			if config.Arch == "aarch64" && strings.Contains(cmdlineString, "ttyS0") {
+				// On aarch64 the serial is ttyAMA0, rewrite command line
+				cmdlineString = strings.Replace(cmdlineString, "ttyS0", "ttyAMA0", 1)
+			}
+			qemuArgs = append(qemuArgs, "-append", cmdlineString)
```

Comment:
@ijc yeah it's skanky
@justincormack I'm not sure I like adding `console=ttyAMA0` to all YAML files but I also don't like what I did above...what do you think about `console=serial` in the YAML and then we replace it in `linuxkit`?